### PR TITLE
i#6897 sched file: Create drmemtace schedule file library

### PR DESCRIPTION
**Pull request by derekbruening**
Refactors the raw2trace and invariant_checker code dealing with schedule files into a common "schedule_file" library.

Adds some unit tests for the library.

Issue: #6897

### CODE REVIEW

**Comment by derekbruening** [comment]
x32 failure is #6303.
The RISC-V failure I don't understand: trying to see if I can reproduce it locally; not clear how it would be caused by this PR so going ahead w/ review but will resolve before merge.

**Comment by derekbruening** [comment]
The RISCV build is failing in master and other PR's too: it is some general issue (prob from a VM update).  I filed #6900.

**Review by abhinav92003 — APPROVED** [review]

File: clients/drcachesim/common/schedule_file.cpp [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+// Returns the uncollapsed record sequence.
+const std::vector<schedule_entry_t> &
+schedule_file_t::get_full_serial_records()
+{
+    if (!aggregated_)
+        aggregate_schedule_data();
+    return serial_;
+}
+// Returns the record sequence with consecutive-same-thread entries collapsed.
```
[the remark this answers is not in the document]
Comment (derekbruening) [reply clients/drcachesim/common/schedule_file.cpp:77]: Done.

**Comment by derekbruening** [comment]
The RISCV failure is #6900.

**Pull request merged**